### DISCUSSION TITLE
Real time Chat App💌

### DISCUSSION THREAD

**Discussion by iamrahul8** [scaleracademy/open-source-projects#96] [Ideas]
The use of messaging apps is growing rapidly. 
This real-time chat app's core idea is collaboration and communication,
which create an effective work environment and culture. 
>It transmits and displays messages to a recipient without the need to reload the website.

### Some of the basic requirements and features : 
- Users create an account by providing a handle that is specific to each user (a name). People are therefore free to select any handle, maintaining their anonymity.

- A member has access to other members' online statuses. They can choose to make their message visible to everyone in the chat room by selecting the "public" option.

- Senders of private messages ought to approach other members first. After accepting the request, other members can talk privately with them.

#### Technology / Tools🛠 used :
- MERN Stack(Mongo, Express, React, Node).
- Socket.io 
- Sign-up and login integration

### Demo 
- https://xanthronschatapp.herokuapp.com

```Do upvote if you like this idea💡```

**Comment by techysubham** [comment]
Good One!
We can Implement this 👍🏻

**Comment by sangita9853** [comment]
For sure a good idea!!

**Comment by Anishgupta0** [comment]
Good idea

**Comment by manmohan4142** [comment]
Nice idea !!

**Comment by rajen1212** [comment]
We can use web-RTC for P2P connection over http.

**Reply by iamrahul8** [reply]
> Comment options

Nice one 🥳
adding Voice&Video Call using P2P WebRTC✅

**Comment by Harshit1020** [comment]
Good One!
We can Implement this .

**Comment by BishalBiswal** [comment]
Nice, good one.

**Comment by kobruhhhhh** [comment]
Nice idea !!

**Comment by imhardikdesai** [comment]
It is a Good Idea. We Can Implement this

**Comment by johnwill0** [comment]
Wao that's great, that really a good idea. I will implement this practice to my service based website [Gulbenkian Andonian Solicitors](https://www.gulbenkian.co.uk/). Actually I am offering law related services here. And my clients text me via these links, sometimes links got expire and this rezult in bad user experience. Now I hope by using your technique I will overcome my this problem.

**Comment by LorraineBenson** [comment]
Oh wao, that's really a great idea. I will implement it on my service based website so that my [clients](https://rip-notices.com/) communicate with me live and may resolve their issues on the spot. Really it will be a great initiative.
